google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -1379,6 +1379,14 @@ public com.google.api.services.bigquery.model.QueryResponse call() {
   public TableResult query(QueryJobConfiguration configuration, JobId jobId, JobOption... options)
       throws InterruptedException, JobException {
     Job.checkNotDryRun(configuration, "query");
+    // If all parameters passed in configuration are supported by the query() method on the backend,
+    // put on fast path
+    QueryRequestInfo requestInfo = new QueryRequestInfo(configuration);
+    if (requestInfo.isFastQuerySupported(jobId)) {
```

Comment:
@Neenu1995 If other attributes in the `config` are set, like Clustering, TimePartitioning etc then wouldn't it take the slow path leading to Already-Exists error. Do we think if doing a `jobs.get` call and avoiding the `create` call when the job already exists could be a good idea here?

Reply:
I realize we can't avoid the slow path always. When the conditions are met for the fast path, then it should take fast path. 
But here, in the query method which passes job id, the only scenario where it can take the fast path is, if the location and jobid fields are not set. In such cases, the job id object is entirely disregarded. I am not sure if that is the right way to do this. What do you think?

Reply:
Logically it is making sense to me as we are using fast query path when the jobId and location isn't passed. Couple of quick questions:

1. Is this fix supposed to be addressing the `Already-exists` issue? I am sorry if I am getting confused with that here. :)
2. Are we already covering this new flow with an Integration Test?